procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+                    * `service_account` set to `$_google_play_accounts['$product']['service_account']` (so it fetches from `hiera`)
+                    * `certificate` set to `$_google_play_accounts['$product']['certificate']` (so it fetches from `hiera`)
+                    * `certificate_target_location` set to `${root}/$product.p12`
+                2. In `$google_play_accounts_config_content`, add a new dictionary at `$product` with properties that point `service_account` and `certificate` to the values in `$google_play_config`
+                3. In `$jarsigner_certificate_aliases_content`, add a mapping from `$product` to the same string: `"$product"`
+            4. In `$jarsigner_all_certificates`
+                1. Add two entries:
+                    * `'$product-dep' => "${root}/$product_dep.cer"`
+                    * `'$product-release' => "${root}/$product_release.cer"`
+        3. In `manifests/init.pp`
+            1. For both `mobile-dep` and `mobile-prod`, add an entry so that at `certificate_target_location`, the `content` is set to `certificate` for your product
+        4. In `manifests/jarsigner_init.pp`, for both `mobile-dep` and `mobile-prod`:
+            1. For convenience, set a variable at the top of the section that points to the relevant certificate location (either `dep` or `prod` depending on the section you're currently on)
+            2. Add an entry to the `file` block so that, at the certificate location, the source of the correct `pem` file is copied in
+            3. Add an entry to the `java_ks` block for your product, setting `certificate` to your certificate location
+    5. In `files/requirements.txt`
+        1. From step 9, update the version of `mozapkpublisher` and `pushapkscript`
+11. Commit and push your `build-puppet` changes, make a PR
+12. Test your automation with staging releases
+    1. Confirm with the team that nobody is using the dep signing or pushapk workers
+    2. SSH into the puppet master
+    3. Go to `/etc/puppet/environments/$username`
+    4. Temporarily move files out of the directory (like `environment.conf`)
+    5. `git clone https://github.com/$username/build-puppet .` to clone `build-puppet` to the directory
+    6. Checkout the branch you just pushed in step 11
+    7. Move the files back into the directory
+    8. Link config and nodes from the "main" environment
+        1. `cd manifests`
+        2. `ln -s moco-config.pp config.pp`
+        3. `ln -s moco-nodes.pp nodes.pp`
+    9. Edit `manifests/moco-nodes.pp` to pin both `dep-m-signing-linux` and `dep-m-pushapkworker` to your environment
+        * Add `$pin_puppet_server = 'releng-puppet2.srv.releng.mdc1.mozilla.com'`
+        * Add `$pin_puppet_env = '$username'`
+    10. SSH into both the pushapk worker (`dep-m-pushapkworker-1.srv.releng.use1.mozilla.com`) and the signing worker (`dep-m-signing-linux-1.srv.releng.use1.mozilla.com`)
```

Comment:
Tip: if we later add more instances and ssh-ing to each of them becomes impracticle, this can also be achieved by using `csshX`.

```
# vpn
for i in {1..10}; do
nslookup dep-m-pushapkworker-$i | grep Name: | sed -e 's/Name:\t//'
done > /tmp/dep-pushapkworker
csshX --hosts /tmp/dep-pushapkworker
sudo puppet agent --test  # or unpin or w/e
```

Reply:
Since these are dep workers, I'm feeling lucky that we'll sit with one worker (at least for a while).
This is worth revisiting if we start adding additional dep workers

Reply:
Makes total sense, I added it more for the future context 👍